batcher/batcher.go

```diff
@@ -158,6 +165,11 @@ func (b *basicBatcher) Dispose() {
 	b.flush()
 	b.disposed = true
 	b.items = nil
+
+	// Drain the batch channel and all routines waiting to put on the channel
+	for len(b.batchChan) > 0 || atomic.LoadInt32(&b.waiting) > 0 {
+		<-b.batchChan
+	}
 	close(b.batchChan)
```

Comment:
So what happens here if 1000 goroutines are waiting to send on this channel?  Does only one get cleared before the rest panic on this close?